### PR TITLE
[Feature-8386][Server] Custom timezone - Add timezone into user info

### PR DESCRIPTION
**Pull request by jim-parsons**
## Purpose of the pull request

Custom timezone - Add timezone into user info

## Brief change log

  1. Add column `time_zone varchar(32) default null` in table `t_ds_user`
  2. Add system default timezone like `GMT+08:00` when create user
  3. Add timeZone property when update user info

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

**Comment by jim-parsons** [comment]
@caishunfeng plz review，thx~

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/UsersServiceImpl.java [hunk 1]
```diff
@@ -205,6 +207,8 @@ public User createUser(String userName,
             queue = "";
         }
         user.setQueue(queue);
+        String defaultTimeZone = ZoneId.systemDefault().getRules().getOffset(Instant.now()).getId();
+        user.setTimeZone("GMT" + defaultTimeZone);
```
Comment (caishunfeng) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/UsersServiceImpl.java]: Is it better to init the user timezone null? If timezone null, use the server default timezone for date convert. What do you think?

File: dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/service/UsersServiceTest.java [hunk 2]
```diff
@@ -269,13 +269,13 @@ public void testUpdateUser() {
         String userPassword = "userTest0001";
         try {
             //user not exist
-            Map<String, Object> result = usersService.updateUser(getLoginUser(), 0, userName, userPassword, "3443@qq.com", 1, "13457864543", "queue", 1);
+            Map<String, Object> result = usersService.updateUser(getLoginUser(), 0, userName, userPassword, "3443@qq.com", 1, "13457864543", "queue", 1, "+8");
```
Comment (caishunfeng) [inline dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/service/UsersServiceTest.java]:
```suggestion
            Map<String, Object> result = usersService.updateUser(getLoginUser(), 0, userName, userPassword, "3443@qq.com", 1, "13457864543", "queue", 1, "Asia/Shanghai");
```

File: dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/service/UsersServiceTest.java [hunk 3]
```diff
@@ -269,13 +269,13 @@ public void testUpdateUser() {
...
             Assert.assertEquals(Status.USER_NOT_EXIST, result.get(Constants.STATUS));
             logger.info(result.toString());
 
             //success
             when(userMapper.selectById(1)).thenReturn(getUser());
-            result = usersService.updateUser(getLoginUser(), 1, userName, userPassword, "32222s@qq.com", 1, "13457864543", "queue", 1);
+            result = usersService.updateUser(getLoginUser(), 1, userName, userPassword, "32222s@qq.com", 1, "13457864543", "queue", 1, "+8");
```
Comment (caishunfeng) [inline dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/service/UsersServiceTest.java]:
```suggestion
            result = usersService.updateUser(getLoginUser(), 1, userName, userPassword, "32222s@qq.com", 1, "13457864543", "queue", 1, "Asia/Shanghai");
```

File: dolphinscheduler-dao/src/main/resources/sql/dolphinscheduler_h2.sql [hunk 4]
```diff
@@ -979,7 +980,7 @@ VALUES ('1,2', 1, 'default admin warning group', 'default admin warning group',
 -- ----------------------------
 INSERT INTO t_ds_user
 VALUES ('1', 'admin', '7ad2410b2f4c074479a8937a28a22b8f', '0', 'xxx@qq.com', '', '0', '2018-03-27 15:48:50',
-        '2018-10-24 17:40:22', null, 1);
+        '2018-10-24 17:40:22', null, 1, 'GMT+8');
```
Comment (caishunfeng) [inline dolphinscheduler-dao/src/main/resources/sql/dolphinscheduler_h2.sql]:
```suggestion
        '2018-10-24 17:40:22', null, 1, null);
```

File: dolphinscheduler-dao/src/main/resources/sql/upgrade/2.0.4_schema/mysql/dolphinscheduler_ddl.sql [hunk 5]
```diff
@@ -14,3 +14,5 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
 */
+
+ALTER TABLE `t_ds_user` ADD COLUMN `time_zone` varchar(4) COMMENT 'time zone, e.g. +8' AFTER `state`;
```
Comment (caishunfeng) [inline dolphinscheduler-dao/src/main/resources/sql/upgrade/2.0.4_schema/mysql/dolphinscheduler_ddl.sql]:
```suggestion
ALTER TABLE `t_ds_user` ADD COLUMN `time_zone` varchar(4) COMMENT 'time zone' AFTER `state`;
```

**Comment by jim-parsons** [comment]
@caishunfeng i‘ve updated the code, thx for your advice, plz take a review

**Review by caishunfeng — APPROVED** [review]
LGTM

**Comment by caishunfeng** [comment]
> @caishunfeng i‘ve updated the code, thx for your advice, plz take a review

Look good to me, I had merged the pr, thanks for your contribution.

**Comment by caishunfeng** [comment]
this pr close #8386

**Pull request merged**